pkg/apis/eventing/v1/broker_lifecycle.go

```diff
@@ -31,6 +33,18 @@ var brokerCondSet = apis.NewLivingConditionSet(
 	BrokerConditionAddressable,
 )
 
+var customConditionSet = map[string]apis.ConditionSet{
+	"":                                 brokerCondSet,
+	eventing.MTChannelBrokerClassValue: brokerCondSet,
+}
+
+// RegisterAlternateBrokerConditionSet register a apis.ConditionSet for the given broker class.
+//
+// Calls to this function need to be synchronized by the caller (not thread-safe).
```

Comment:
If the calls to this need to be synchronized, don't you also need to synchronize the GetConditions?

Reply:
I added an `RWMutex`.